### PR TITLE
init

### PR DESCRIPTION
**Pull request by adminoren**

### CODE REVIEW

**Comment by yecli** [comment]
@OlegoO Could someone from training team prepare module functionality description, like for other modules? That would be good starting point for reviewing functionality.

File: VirtoCommerce.CustomerReviews.Data/Services/CustomerReviewSearchService.cs [hunk 1]
```diff
@@ -49,10 +49,12 @@ public async Task<GenericSearchResult<CustomerReview>> SearchCustomerReviewsAsyn
...
                 }
 
-                if (criteria.ReviewStatus.HasValue)
+                if (!criteria.ReviewStatus.IsNullOrEmpty())
                 {
-                    var convertedStatus = (byte)criteria.ReviewStatus.Value;
-                    query = query.Where(x => x.ReviewStatus == convertedStatus);
+                    foreach(var statusId in criteria.ReviewStatus)
+                    {
+                        query = query.Where(r => r.ReviewStatus == (byte)statusId);
```
Comment (eMazeika) [inline VirtoCommerce.CustomerReviews.Data/Services/CustomerReviewSearchService.cs]: will this allow to search by multiple statuses?

Comment (adminoren) [reply VirtoCommerce.CustomerReviews.Data/Services/CustomerReviewSearchService.cs]: Thanks for your review! Of course, this code doesn't work with multiple statuses. The field reviewStatus in entity model was byte, and when I casted reviewStatuses collection of int to byte and used in  "Contains()..." it didn't work. When I used reviewStatuses collection of int the result sql code contained cast from int to byte and it prevented using index. I have changed the type of field ReviewStatus to int and fixed the highlighted code.

File: VirtoCommerce.CustomerReviews.Core/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (yecli) [inline VirtoCommerce.CustomerReviews.Core/Properties/AssemblyInfo.cs]: For versioning and other common assembly info parts we could use `CommonAssemblyInfo` [example](https://github.com/VirtoCommerce/vc-module-catalog/blob/dev/CommonAssemblyInfo.cs)

File: VirtoCommerce.CustomerReviews.Web/Scripts/VirtoCommerce.CustomerReviews.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    function ($rootScope, mainMenuService, widgetService, $state, authService) {
+
+        //Register reviews menu item
+        var menuItemNewReviews = {
+            path: 'browse/customerReviewsNew',
+            icon: 'fa fa-comments',
+            title: 'Customer Reviews',
+            priority: 100,
+            action: function () { $state.go('workspace.customerReviews') },
+            permission: 'customerReviews:read'
```
Comment (yecli) [inline VirtoCommerce.CustomerReviews.Web/Scripts/VirtoCommerce.CustomerReviews.js]: Probably need 'customerReviews:access' permission here.

File: VirtoCommerce.CustomerReviews.Web/Scripts/blades/review-detail.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        function saveChanges() {
+            blade.isLoading = true;
+            reviewsAPI.update({}, [blade.currentEntity],
+                function (data, headers) {
+                    blade.isLoading = false;
+                    blade.parentBlade.refresh(true);
+                    blade.origEntity = blade.currentEntity;
+                    $scope.bladeClose();
+                },
+                function (error) {
```
Comment (yecli) [inline VirtoCommerce.CustomerReviews.Web/Scripts/blades/review-detail.js]: Probably no need in error hanlders as platform httpInterceptor could handle it.

File: VirtoCommerce.CustomerReviews.Web/Scripts/blades/review-detail.tpl.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        <div class="inner-block">
+            <form class="form" name="formScope">
+                <div ng-init="setForm(formScope)">
+                    <label class="form-label">{{'customerReviews.blades.review-detail.labels.productName' | translate}}</label>
+                    <div class="form-input">
+                        <input class="__small" ng-model="blade.currentEntity.productName" focus-on="" required rows="1" disabled="disabled">
+                    </div>
+                </div>
+
+                <div ng-init="setForm(formScope)">
```
Comment (yecli) [inline VirtoCommerce.CustomerReviews.Web/Scripts/blades/review-detail.tpl.html]: Isn't one ng-init with setForm enough?

File: VirtoCommerce.CustomerReviews.Web/Scripts/blades/review-detail.tpl.html [hunk 6]
```diff
@@ -0,0 +1,55 @@
+<div class="blade-content blade-content __large-wide">
+    <div class="blade-inner">
+        <div class="inner-block">
```
Comment (yecli) [inline VirtoCommerce.CustomerReviews.Web/Scripts/blades/review-detail.tpl.html]: Probably we need to think about extensibility: va-metaform or something.

File: VirtoCommerce.CustomerReviews.Data/Repositories/CustomerReviewRepository.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+        public Task<CustomerReviewEntity[]> GetByIdsAsync(IEnumerable<string> ids)
+        {
+            return CustomerReviews.Where(x => ids.Contains(x.Id)).ToArrayAsync();
+        }
+
+        public async Task DeleteCustomerReviewsAsync(IEnumerable<string> ids)
+        {
+            var items = await GetByIdsAsync(ids);
+            foreach (var item in items)
```
[the remark this answers is not in the document]
Comment (kamilgarey) [reply VirtoCommerce.CustomerReviews.Data/Repositories/CustomerReviewRepository.cs]: We need to use third party [libraries ](https://entityframework.net/batch-delete) for this. I think it's not so perfomance critical to add an library for.

File: VirtoCommerce.CustomerReviews.Data/Repositories/CustomerReviewRepository.cs [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        public IQueryable<RatingEntity> Ratings => GetAsQueryable<RatingEntity>();
+
+        public async Task<RatingEntity[]> GetAsync(string storeId, IEnumerable<string> productIds)
+        {
+            return await Ratings
+                .Where(x => x.StoreId == storeId && productIds.Contains(x.ProductId))
+                .ToArrayAsync();
+        }
+
+        public async Task<RatingEntity> GetAsync(string storeId, string productId)
```
[the remark this answers is not in the document]
Comment (kamilgarey) [reply VirtoCommerce.CustomerReviews.Data/Repositories/CustomerReviewRepository.cs]: fathermore this method is not used anywhere

**Comment by akak1977** [comment]
It seems this is test example

**Pull request closed without merging**